core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/ImageComponentUpdaterService.java

```diff
@@ -44,21 +42,39 @@ public class ImageComponentUpdaterService {
     @Inject
     private CloudbreakRestRequestThreadLocalService restRequestThreadLocalService;
 
-    public UpgradeImageInfo updateForUpgrade(String targetImageId, Long stackId) {
+    public UpgradeImageInfo updateComponentsForUpgrade(String targetImageId, Long stackId) {
         Stack stack = stackService.getById(stackId);
+        restRequestThreadLocalService.setWorkspace(stack.getWorkspace());
+        UpgradeImageInfo upgradeImageInfo = getUpgradeImageInfo(targetImageId, stackId, stack);
+        updateComponents(upgradeImageInfo.getTargetStatedImage(), stack, targetImageId);
+
+        return upgradeImageInfo;
+    }
+
+    public void updateComponentsForUpgrade(StatedImage targetStatedImage, Long stackId) {
+        Stack stack = stackService.getById(stackId);
+        restRequestThreadLocalService.setWorkspace(stack.getWorkspace());
+        updateComponents(targetStatedImage, stack, targetStatedImage.getImage().getUuid());
+    }
```

Comment:
one more small thing: would it make sense to add log messages to these methods? I think it would help later